### PR TITLE
Move FileDropZone to core

### PR DESCRIPTION
**Pull request by Fercas123**

### CODE REVIEW

File: packages/core/stories/file-drop-zone/file-drop-zone.qa.stories.tsx [hunk 1]
```diff
@@ -2,12 +2,12 @@ import {
...
   FileDropZoneIcon,
   FileDropZoneTrigger,
-} from "@salt-ds/lab";
+} from "@salt-ds/core";
 import { Meta, StoryFn } from "@storybook/react";
 import { AllRenderer } from "docs/components";
 
 export default {
-  title: "Lab/File Drop Zone/QA",
+  title: "Core/File Drop Zone/QA",
```
Comment (feliciackh) [inline packages/core/stories/file-drop-zone/file-drop-zone.qa.stories.tsx:10]:
```suggestion
  title: "Core/File Drop Zone/File Drop Zone QA",
```
for easier recognition on chromatic :D

File: .changeset/hungry-pots-guess.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+---
+"@salt-ds/core": minor
+---
+
+Added `FileDropZone` to Core
```
Comment (joshwooding) [inline .changeset/hungry-pots-guess.md:5]:
```suggestion
Added `FileDropZone`, `FileDropZoneIcon` and `FileDropZoneTrigger` to Core.
```

File: site/src/examples/file-drop-zone/Error.tsx [hunk 3]
```diff
@@ -1,10 +1,11 @@
 import { DragEvent, ReactElement } from "react";
-import { Text } from "@salt-ds/core";
 import {
+  Text
```
Comment (lukeac123) [inline site/src/examples/file-drop-zone/Error.tsx:3]:
```suggestion

```

**Review by joshwooding — APPROVED** [review]

File: .changeset/four-windows-beam.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"@salt-ds/lab": minor
+---
+
+Removed `FileDropZone` from Lab and promoted it to Core.
```
Comment (alycrys) [inline .changeset/four-windows-beam.md:5]:
```suggestion
Removed `FileDropZone` from Lab and promoted it to core.
```

Comment (Fercas123) [reply .changeset/four-windows-beam.md]: do we have a consensus about it? I see both being used in the release notes, but this makes sense given how we talk about them in the site... updated :)

**Review by alycrys — APPROVED** [review]

**Review by joshwooding — APPROVED** [review]

**Pull request merged**